rack-ui/app.py

```diff
@@ -0,0 +1,209 @@
+                    tmp_dir = "/tmp/ingestion_package_uploaded_" + datetime.now().strftime("%Y%m%d-%H%M%S")  # temp directory to store the unzipped package
+                    content_type, content_string = content.split(',')
+                    zip_str = io.BytesIO(base64.b64decode(content_string))
+                    zip_obj = ZipFile(zip_str, 'r')
+                    zip_obj.extractall(path=tmp_dir)  # unzip the package
+                    default_manifest_file_name = "manifest.yaml"  # look for this manifest file
+                    manifests = glob.glob(tmp_dir + '/**/' + default_manifest_file_name, recursive=True)
+                    if len(manifests) == 0:
+                        raise Exception("Cannot load ingestion package: does not contain a default manifest file (" + default_manifest_file_name + ")")
+                    if len(manifests) > 1:
+                        raise Exception("Cannot load ingestion package: contains multiple default manifest files: " + str(manifests))
+                    manifest = manifests[0]
+                    rack.ingest_manifest_driver(Path(manifest), BASE_URL, TRIPLE_STORE, TRIPLE_STORE_TYPE, True)  # process the manifest
+            return True, "Loaded ingestion package using manifest file " + manifest[(len(tmp_dir) + 1):]  # show modal dialog with success message
+        except Exception as e:
+            return True, get_error_trace(e)  # show modal dialog with error
+    elif modal_is_open and n_clicks_close is not None and n_clicks_close > 0:  # user clicked close on modal dialog
+        in_mem_file.truncate(0)  # clear status
+        in_mem_file.seek(0)  # clear status
+    return False, ""  # hide (or don't show) modal dialog
+
+# @app.callback([Output('modal-reset', 'is_open'), Output('div-modal-reset', 'children')],
+#               Input('button-reset-provider', 'submit_n_clicks'),  # user clicked "OK" on confirmation dialog
+#               Input('button-modal-reset-close', 'n_clicks'),
+#               State('modal-reset', 'is_open'),
+#               prevent_initial_call=True)
+# def reset(n_clicks_reset, n_clicks_close, modal_is_open):
+#     """ Callback triggered when user selects reset """
+#     if not modal_is_open and n_clicks_reset is not None and n_clicks_reset > 0:
+#         try:
+#             with redirect_stdout(in_mem_file):
+#                 # load RACK
+#                 rack.ingest_manifest_driver(Path("../cli/rack.yaml"), BASE_URL, TRIPLE_STORE, TRIPLE_STORE_TYPE, True)
+#             return True, "RACK has been reset"  # show modal dialog
+#         except Exception as e:
+#             return True, get_error_trace(e)  # show modal dialog with error
+#     elif modal_is_open and n_clicks_close is not None and n_clicks_close > 0:  # user clicked close on modal dialog
+#         in_mem_file.truncate(0)  # clear status
+#         in_mem_file.seek(0)  # clear status
+#     return False, ""  # hide (or don't show) modal dialog
```

Comment:
If you don't plan to uncomment this code later, best practice is remove all dead code to avoid cluttering greps hits, searches, etc.